bitnami/spark/templates/hpa-worker.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.worker.enableAutoscaling }}
```

Comment:
I would use `.Values.worker.autoscaling.enabled`.

That way you already have a config section specific for this hpa and you can reuse it for other configurations. For example, in this template you are also using:

`.Values.hpa.resource` , while I would use `.Values.worker.autoscaling.resource`